plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImpl.java

```diff
@@ -839,7 +848,7 @@ private void validateManagedKubernetesClusterCreateParameters(final CreateKubern
     }
 
     private Network getKubernetesClusterNetworkIfMissing(final String clusterName, final DataCenter zone,  final Account owner, final int controlNodesCount,
-                         final int nodesCount, final String externalLoadBalancerIpAddress, final Long networkId) throws CloudRuntimeException {
+                         final int nodesCount, final String externalLoadBalancerIpAddress, final Long networkId, final Long asNumber) throws CloudRuntimeException {
```

Comment:
@nvazquez @Pearl1594 
asnumber is a parameter of network, not CKS cluster. 
if we decide to accept `asnumber` as a CKS parameter, should we also accept other network parameters like network offering, network cidr, network service providers ?
In my point of view, if user want to create CKS cluster on networks with non-default network offering, they should create the network in advanced, then create CKS cluster on it.

Reply:
it looks like thanges in `plugins/integrations/kubernetes-service/` are not needed.

Reply:
So, when we use a pre-existing network to deploy a CKS cluster, and the network has an offering that accepts AS number ,then we wouldn't pass as number during CKS cluster creation. But, when we set a network offering in the global setting that requires an AS number (i.e., specify AS number is true) and we want to create an isolated network during CKS cluster creation, then we rely on the AS number to be passed during the cluster creation phase. This would ultimately be used when creating the network. So effectively, AS number is a parameter of the network and not CKS .

Reply:
@Pearl1594 
I understood. However, I still think this is not needed
if specifyvlan is true, do we pass the vlan in the CKS cluster creation ?
in the future, if there are more network parameters are supported, will we add them to the CKS cluster creation API ?
In my opinion, we should ask users to create network prior to CKS creation, which provides more options.

Reply:
Ok, makes sense .. we could probably add a alert / warning, if the network offering set has specify AS and network id isn't passed, if we do not want AS number to be passed at the cluster creation phase. cc @nvazquez

Reply:
yes, the checks I can remember are
- on AS number if specifyasnumber is true
- on vlan if specifyvlan is true
- on gateway/netmask if network mode is ROUTED

there may be more.
it would be better if we can re-use the checks in 
https://github.com/apache/cloudstack/blob/d41dc17b2c2d89c878472c636aa38bb015950927/server/src/main/java/com/cloud/network/NetworkServiceImpl.java#L1441-L1750

Reply:
Ok @weizhouapache @Pearl1594 - we can revert the changes as they were to assign an AS number only in case network was not provided and CKS needed to create it, and also the offering supported Specify AS number and Dynamic Routing